### PR TITLE
Convert to use the electron-installer-common ElectronInstaller class

### PR DESCRIPTION
**Pull request by malept**
Utilizes `electron-installer-common` 0.6.0. This is a PR to merge into #244.

### CODE REVIEW

File: src/installer.js [hunk 1]
```diff
@@ -21,261 +21,231 @@ function defaultRename (dest, src) {
...
-    createSpec,
-    createApplication
-  ])
-}
+  /**
+   * Get the hash of default options for the installer. Some come from the info
+   * read from `package.json`, and some are hardcoded.
+   */
+  generateDefaults () {
+    return common.readMeta(this.userSupplied)
```
Comment (fcastilloec) [inline src/installer.js]: This is the reason why it's failing on all CIs
```suggestion
    return common.readMetadata(this.userSupplied)
```

File: src/installer.js [hunk 2]
```diff
@@ -21,261 +21,231 @@ function defaultRename (dest, src) {
...
-      data.logger(`Successfully created package at ${options.dest}`)
-      return options
+      data.logger(`Successfully created package at ${installer.options.dest}`)
+      return installer.options
     }).catch(err => {
       data.logger(common.errorMessage('creating package', err))
       throw err
     })
 
   return nodeify(promise, callback)
```
Comment (fcastilloec) [inline src/installer.js]: As with `-redhat` we also need to remove callbacks from here.

File: src/installer.js (hunk 1 above)
Comment (malept) [reply src/installer.js]: Yup, fixed it elsewhere but forgot it was used here.

File: src/installer.js (hunk 2 above)
Comment (malept) [reply src/installer.js]: We should do that in a separate PR (in both modules).

Comment (fcastilloec) [reply src/installer.js]: Ok, sounds good

**Comment by fcastilloec** [comment]
@malept we should also just build master and PR for this module. Otherwise, we're having too many duplicates. This is particularly important on AppVeyor because it doesn't do parallel builds.
Based on [this post](https://help.appveyor.com/discussions/questions/2325-building-only-pull-requests-and-master-branch) it seems that we only need to add:

```yml
branches:
  only:
  - master
```

**Comment by malept** [comment]
We also want tagged releases.

**Comment by malept** [comment]
I'll do the restricted branch change in a different PR.

**Pull request merged**